### PR TITLE
Monero wallet rpc installer and executor

### LINKED ISSUE #224: Monero wallet RPC management
- Download Monero wallet RPC for user into temp directory
- Start with correct parameters
- Stop when terminated

### PR DESCRIPTION
**Pull request by rishflab**
Fixes #224

### CODE REVIEW

File: swap/src/monero/wallet_rpc.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+};
+
+#[cfg(target_os = "macos")]
+const DOWNLOAD_URL: &str = "http://downloads.getmonero.org/cli/monero-mac-x64-v0.17.1.9.tar.bz2";
+
+#[cfg(target_os = "linux")]
+const DOWNLOAD_URL: &str = "https://downloads.getmonero.org/cli/monero-linux-x64-v0.17.1.9.tar.bz2";
+
+#[cfg(not(any(target_os = "macos", target_os = "linux")))]
+compile_error!("unsupported operating system");
```
Comment (thomaseizinger) [inline swap/src/monero/wallet_rpc.rs:29]: Can you also open an issue for Windows support please?

**Review by thomaseizinger — APPROVED** [review]
Thanks for putting the PR up, good work today!

**Review by da-kami — APPROVED** [review]
Noice!

**Pull request merged**